include/seqan3/io/alignment_file/input.hpp

```diff
@@ -875,6 +878,8 @@ class alignment_file_input
      */
     //!\brief Buffer for a single record.
     record_type record_buffer;
+    //!\brief A larger (compared to stl default) stream buffer to use when reading from a file.
+    char stream_buffer[1'000'000];
```

Comment:
Not sure if we want a stack buffer of that size. Did you double check how seqan2 did this?

Reply:
@rrahn said this is what seqan2 is doing (1 MB buffer) :woman_shrugging: but I did not confirm again. I also wanted to have this PR so we can talk about it but I thought it is good if somebody already has a close look at it because I was unsure if that's the way to do it.

Reply:
Yes, that's also what I suggested. Stack buffer is completely fine, but I would still use `std::array` because that's always preferable.

Reply:
I changed it to a heap buffer since the performance did not drop by the change.

Reply:
So I looked more closely into the SeqAn implementation and it seems that the default buffer size changes quite a lot for different stream implementations. But nonethless I believe using heap or stack is not important here. But I am in favour of heap memory at this point.